datafusion/core/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -323,8 +318,17 @@ pub(crate) struct SortPreservingMergeStream {
     /// An id to uniquely identify the input stream batch
     next_batch_id: usize,
 
-    /// Heap that yields [`SortKeyCursor`] in increasing order
-    heap: BinaryHeap<Reverse<SortKeyCursor>>,
+    /// Vector that holds all [`SortKeyCursor`]s
+    cursors: Vec<Option<SortKeyCursor>>,
+
+    /// The loser tree that always produces the minimum cursor
+    ///
+    /// Node 0 stores the top winner, Nodes 1..num_streams store
+    /// the loser nodes
+    loser_tree: Vec<usize>,
+
+    /// Identify whether the loser tree is adjusted
```

Comment:
```suggestion
    /// Identify whether the most recently yielded overall winner has been replaced
    /// within the loser tree, a value of `false` indicates that they overall winner
    /// has been yielded but the loser tree has not been updated
```

Or something to make it clearer what adjusted actually means.

FWIW a boolean of `should_replace_winner` or something might be clearer